src/zenml/zen_stores/schemas/step_run_schemas.py

```diff
@@ -222,10 +228,7 @@ def to_model(
             RuntimeError: If the step run schema does not have a deployment_id
                 or a step_configuration.
         """
-        run_metadata = {
-            metadata_schema.key: json.loads(metadata_schema.value)
-            for metadata_schema in self.run_metadata
-        }
+        run_metadata = self.fetch_metadata()
```

Comment:
Can we fetch this inside the `if`, this is only needed when `include_metadata` is True>

Reply:
Same for other the pipeline run

Reply:
Good catch! Fixed this.